### PR TITLE
fix kubectl describe ingress shows default-backend:80 when no default…

### PR DESCRIPTION
**Pull request by jlsong01**
#### What type of PR is this?

/kind bug
/sig cli

#### What this PR does / why we need it:
kubectl describe ingress shows default-backend:80 when no default backend is present

#### Which issue(s) this PR fixes:

Fixes #107720
https://github.com/kubernetes/kubectl/issues/1172

cc @soltysh 

#### Does this PR introduce a user-facing change?

```release-note
print <default> as the value in case kubectl describe ingress shows default-backend:80 when no default backend is present
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/describe/describe.go [hunk 1]
```diff
@@ -2640,17 +2640,29 @@ func (i *IngressDescriber) describeIngressV1(ing *networkingv1.Ingress, events *
...
+			defaultBackend := &networkingv1.IngressBackend{
 				Service: &networkingv1.IngressServiceBackend{
 					Name: "default-http-backend",
 					Port: networkingv1.ServiceBackendPort{
 						Number: 80,
 					},
 				},
 			}
-			ns = metav1.NamespaceSystem
+			defaultNamesapce := metav1.NamespaceSystem
```
Comment (lianghao208) [inline staging/src/k8s.io/kubectl/pkg/describe/describe.go:2651]: IMO, `defaultNamesapce` may confuse with `metav1.NamespaceDefault`.
We can just use `metav1.NamespaceSystem` directly or replace it with `systemNamespace`, WDYT?
```
_, err := i.client.CoreV1().Endpoints(metav1.NamespaceSystem).Get(context.TODO(), defaultBackend.Service.Name, metav1.GetOptions{})
```
```
systemNamesapce := metav1.NamespaceSystem
_, err := i.client.CoreV1().Endpoints(systemNamesapce).Get(context.TODO(), defaultBackend.Service.Name, metav1.GetOptions{})
```

Comment (jlsong01) [reply staging/src/k8s.io/kubectl/pkg/describe/describe.go]: Agree and fixed. Thanks.

File: staging/src/k8s.io/kubectl/pkg/describe/describe.go [hunk 2]
```diff
@@ -2701,13 +2712,24 @@ func (i *IngressDescriber) describeIngressV1beta1(ing *networkingv1beta1.Ingress
...
-			ns = metav1.NamespaceSystem
+			_, err := i.client.CoreV1().Endpoints(metav1.NamespaceSystem).Get(context.TODO(), defaultBackend.ServiceName, metav1.GetOptions{})
+			if err == nil {
+				_, err := i.client.CoreV1().Services(metav1.NamespaceSystem).Get(context.TODO(), defaultBackend.ServiceName, metav1.GetOptions{})
+				if err == nil {
+					def = defaultBackend
+					ns = metav1.NamespaceSystem
+				}
+			}
+		}
```
Comment (wangyysde) [inline staging/src/k8s.io/kubectl/pkg/describe/describe.go:2727]: Is it right that defaultBackend will be showed if the endpoint named "default-http-backend" and the service named "default-http-backend" are exist in the metav1.NamespaceSystem when "def(ing.Spec.Backend)" is nil?  But I think that users will be confused  in the following case:
1. there are many ingresses in a namespace;
2. the endpoint named "default-http-backend" and the service named "default-http-backend" are exist in the namespace; 
3. but the above endpoint and service is not prepared for the ingress that the user described(it is prepared for another ingress).

So I think that defaultBackend should  be showed as  "none" when ing.Spec.Backend is nil.

File: staging/src/k8s.io/kubectl/pkg/describe/describe.go [hunk 3]
```diff
@@ -2640,17 +2640,28 @@ func (i *IngressDescriber) describeIngressV1(ing *networkingv1.Ingress, events *
...
+			defaultBackend := &networkingv1.IngressBackend{
 				Service: &networkingv1.IngressServiceBackend{
 					Name: "default-http-backend",
 					Port: networkingv1.ServiceBackendPort{
 						Number: 80,
 					},
 				},
 			}
-			ns = metav1.NamespaceSystem
+			_, err := i.client.CoreV1().Endpoints(metav1.NamespaceSystem).Get(context.TODO(), defaultBackend.Service.Name, metav1.GetOptions{})
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/describe/describe.go:2651]: That's not what I requested, in https://github.com/kubernetes/kubernetes/issues/107720#issuecomment-1058224411 I've meant to only print `<default>` when we there is no default backend specified without checking for anything. As mentioned in https://github.com/kubernetes/kubectl/issues/1172#issuecomment-1026135953 it's a convention name, so I'd prefer to only print `<default>` in those cases.

**Review by soltysh — CHANGES_REQUESTED** [review]

File: staging/src/k8s.io/kubectl/pkg/describe/describe.go (hunk 3 above)
Comment (jlsong01) [reply staging/src/k8s.io/kubectl/pkg/describe/describe.go]: Sorry about that. Fixed.

File: staging/src/k8s.io/kubectl/pkg/describe/describe.go (hunk 2 above)
Comment (jlsong01) [reply staging/src/k8s.io/kubectl/pkg/describe/describe.go]: Agress, fixed as showing <default> in case ing.Spec.Backend is nil.

**Comment by jlsong01** [comment]
@soltysh The change request is done. Could you please take a look again?

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/triage accepted
/priority backlog

**Pull request merged**